tools/cli/adminDBScanCommand.go

```diff
@@ -157,3 +168,144 @@ func checkExecution(
 
 	return execution, invariant.NewInvariantManager(ivs).RunChecks(ctx, execution)
 }
+
+// AdminDBScanUnsupportedWorkflow is to scan DB for unsupported workflow for a new release
+func AdminDBScanUnsupportedWorkflow(c *cli.Context) {
+	rps := c.Int(FlagRPS)
+	outputFile := getOutputFile(c.String(FlagOutputFilename))
+	startShardID := c.Int(FlagLowerShardBound)
+	endShardID := c.Int(FlagUpperShardBound)
+
+	defer outputFile.Close()
+	for i := startShardID; i <= endShardID; i++ {
+		listExecutionsByShardID(c, i, rps, outputFile)
```

Comment:
I kind of feel like if the scan is done in only goroutine, it will take a long time to finish and the rps limit is kind of useless and will never be reached.